src/utils.ts

```diff
@@ -128,3 +130,153 @@ export function aggregationToPresentation(aggregation: Aggregation): Presentatio
   }
   return result;
 }
+
+export function replaceVariables(
+  requestSpec: Partial<RequestSpec> | undefined,
+  scopedVars?: ScopedVars
+): Partial<RequestSpec> {
+  const result: typeof requestSpec = {};
+  const t = getTemplateSrv();
+  if (requestSpec === undefined) {
+    return result;
+  }
+  (Object.keys(requestSpec) as Array<keyof Partial<RequestSpec>>).forEach(function <
+    T extends keyof Partial<RequestSpec>
+  >(key: T) {
+    try {
+      const value = requestSpec[key];
+      if (value === undefined) {
+        return;
+      } else if (typeof value === 'string') {
+        result[key] = t.replace(value, scopedVars) as RequestSpec[T];
+      } else if (key === 'host_labels' && Array.isArray(value)) {
+        const labels = [];
+        for (const label of value) {
+          labels.push(t.replace(label, scopedVars));
+        }
+        result['host_labels'] = labels;
+      } else if (key === 'host_tags' && Array.isArray(value)) {
+        const tags = [];
+        for (const tag of value) {
+          tags.push({
+            group: t.replace(tag.group, scopedVars),
+            tag: t.replace(tag.tag, scopedVars),
+            operator: tag.operator,
+          });
+        }
+        result['host_tags'] = tags as [TagValue, TagValue, TagValue];
+      } else if (typeof value === 'object' && 'value' in value) {
+        // NegatableOption
+        result[key] = { ...value, value: t.replace(value.value, scopedVars) } as RequestSpec[T];
+      } else {
+        result[key] = value;
+      }
+    } catch (e) {
+      console.log(e);
```

Comment:
What error could happen here? Should we inform the user?

Reply:
this looks like a debugging artifact, i removed it.